substratest/client.py

```diff
@@ -14,13 +14,15 @@ class Client:
     Parses responses from server to return Asset instances.
     """
 
-    def __init__(self, backend, node_id=None, address=None, user=None, password=None):
+    def __init__(self, debug, node_id=None, address=None, token=None, user=None, password=None):
         super().__init__()
 
         self.node_id = node_id
-        self.backend = backend
-        self._client = substra.Client(backend=backend, url=address, version="0.0", insecure=False)
-        self._client.login(user, password)
+        self.debug = debug
+        self._client = substra.Client(debug=debug, url=address, version="0.0", insecure=False, token=token)
+        if not token:
+            self._client.login(user, password)
+        self.token = self._client._token
```

Comment:
We should avoid accessing a private attribute.

If I remember well the token is outputted by the login function. We could maybe use the following code instead?
```python
if not token:
    token = self._client.login()
self.token = token
```

Reply:
Good catch !